packages/core/src/components/tooltip/tooltip.tsx

```diff
@@ -114,13 +127,43 @@ export class Tooltip<
         this.popoverRef.current?.reposition();
     }
 
+    protected validateProps(props: TooltipProps<T>) {
+        const childrenCount = React.Children.count(props.children);
+        if (childrenCount > 1) {
+            console.warn(Errors.POPOVER_WARN_TOO_MANY_CHILDREN);
+        }
+        // all other warnings should occur in Popover, not here.
+    }
+
     // any descendant ContextMenus may update this ctxState
     private renderPopover = (ctxState: TooltipContextState) => {
-        const { children, compact, disabled, intent, popoverClassName, ...restProps } = this.props;
+        const { children, content, renderTarget, compact, disabled, intent, popoverClassName, ...restProps } =
+            this.props;
+
         const popoverClasses = classNames(Classes.TOOLTIP, Classes.intentClass(intent), popoverClassName, {
             [Classes.COMPACT]: compact,
         });
 
+        const contentElement = Utils.ensureElement(content);
+        const tooltipId = contentElement?.props?.id ?? React.useMemo(() => Utils.uniqueId("tooltip"), []);
```

Comment:
The use of `React.useMemo` here violates the [rules of hooks](https://react.dev/reference/react/useMemo#caveats) since it is called conditionally within the ternary.

Can we extract this default memoized value out to the top level of this component?